packages/yourloops/test/integration/pages/device/device-page.caregiver.spec.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022-2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { act } from '@testing-library/react'
+import { mockAuth0Hook } from '../../mock/auth0.hook.mock'
+import { mockDataAPI, pumpSettingsData } from '../../mock/data.api.mock'
+import { mockNotificationAPI } from '../../mock/notification.api.mock'
+import { patient1Id } from '../../data/patient.api.data'
+import { mockDirectShareApi } from '../../mock/direct-share.api.mock'
+import { renderPage } from '../../utils/render'
+import { mockUserApi } from '../../mock/user.api.mock'
+import { mockPatientApiForCaregivers } from '../../mock/patient.api.mock'
+import { checkDeviceSettingsContent, checkNavigationToDailyView } from '../../assert/device-page.assert'
+import { mockWindowResizer } from '../../mock/window-resizer.mock'
+import { UserRole } from '../../../../lib/auth/models/enums/user-role.enum'
+
+describe('Device page for Caregiver', () => {
+  const firstName = 'Caregiver firstName'
+  const lastName = 'Caregiver lastName'
+
+  beforeEach(() => {
+    mockWindowResizer()
+    mockAuth0Hook(UserRole.Caregiver)
+    mockNotificationAPI()
+    mockDirectShareApi()
+    mockUserApi().mockUserDataFetch({ firstName, lastName })
+    mockPatientApiForCaregivers()
+    mockDataAPI(pumpSettingsData)
+  })
+
+  it('should render correct components when navigating to the device page', async () => {
+    let router
+    await act(async () => {
+      router = renderPage(`/patient/${patient1Id}/device`)
+    })
+    checkDeviceSettingsContent()
```

Comment:
This does not follow the new method naming/architecture that we decided for ITs.
You should call a method name `testDeviceSettingVisualisation` (something like that) defined in a file `use-cases/device-settings-visualisation.ts` that call the method `checkDeviceSettingsContent`.
You could also call `testAppMainLayoutForCaregiver` in here.

Similar comment for the method `checkNavigationToDailyView`.

Alternatively, you can have one `it` per `testSomething` called.

```
it('should render correct layout', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    await testAppMainLayoutForCaregiver()
  })

  it('should display correct parameters', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    testDeviceSettingsContent()
    await checkNavigationToDailyView(router, `/patient/${patient1Id}/daily`)
  })

  it('should navigate to daily page when clicking on the daily button', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    await testNavigationToDailyView(router, `/patient/${patient1Id}/daily`)
  })
```